nodes/src/nodes/library/image/embed.py

```
@@ -58,8 +59,12 @@ def embed_images_in_html(html_content: str, default_scheme: str, default_site: s
             # Download the image
             debug(f'Processing image URL: {image_url}')
 
+            # SSRF protection: validate and pin to resolved IP to prevent TOCTOU
+            _url, hostname, resolved_ips = validate_url(image_url)
+            safe_url, host_headers = build_ssrf_safe_url(image_url, hostname, resolved_ips)
+
             # Check if the image URL is valid
-            response = requests.get(image_url)
+            response = requests.get(safe_url, headers=host_headers, allow_redirects=False)
```

Ruff (0.15.7)

[error] Probable use of `requests` call without timeout

(S113)